### ISSUE TITLE
Cannot use `list[Serializer]` to model root JSON element in `extend_schema`

### ISSUE THREAD

**Issue by johnthagen** [tfranzel/drf-spectacular#921]
I would like to decorate an `@extend_schema` this way:

```py
@extend_schema(responses=list[MySerializer])
```

To publish that my endpoint returns a `list` at its root, and within that, each element is modeled by `MySerializer`.

But this raises an error

```
Warning [my_route]: could not resolve "list[MySerializer]"
```

I can't seem to model this with a normal `Serializer`, as mentioned in this discussion:

- https://github.com/encode/django-rest-framework/discussions/8851

Is there any other way to properly model the root element of the JSON response as a list when it doesn't have a attribute name?

**Comment by tfranzel** [comment]
Hey John,

do you - by any chance - just mean 

```python
@extend_schema(responses=MySerializer(many=True))
```
?

**Comment by johnthagen** [comment]
🤯 

I didn't realize you could pass an instance of a `Serializer` (and also that you could then use `many=True` like that inline like that). It works perfectly!

For the future user, do you think we could add something about this to the docs?

- https://drf-spectacular.readthedocs.io/en/latest/customization.html#step-2-extend-schema

I know we don't want to bog things down too much, but maybe showing an example with an instance like this that has parameters could be useful.

If you don't think so, happy to close this issue. Either way, thank you! ❤️

**Comment by tfranzel** [comment]
It is documented in `extend_schema` and 

https://github.com/tfranzel/drf-spectacular/blob/24b8a9890d59900d102490a48e286d08cdcc7ab5/drf_spectacular/utils.py#L289

and

**Comment by tfranzel** [comment]
Interesting how you didn't use the common DRF idiom relating to lists:

```python

class XS(serializer.Serializer):
    xx = FooSerializer(many=True)
    yy = IntegerField()
```

To me it seemed logical to design it like that. There are a couple more cases where instances come in handy. Classes will be init'ed somewhere along the way anyway unless instances were provided.

This is actually relevant to know when you write `SerializerExtensions`. `self.target` is the instance and `self.target_class` is the class

**Comment by johnthagen** [comment]
> Interesting how you didn't use the common DRF idiom relating to lists

In my case, I'm actually using a custom function-based endpoint that is getting the data through another mechanism, but I'm only using the Serialiazer itself as a way to model the (otherwise opaque) JSON that I am serving back. I am using the serializer as a way to model the data for the OpenAPI representation. It's somewhat uncommon and I don't do this in most of the app (most use normal Serializers with `many=True` etc and pull from the database).

In these more rare situations, I just wasn't immediately obvious to me that you could pass an _instance_ of a `Serializer` to `extend_schema`, but as you mention this is documented deeper in the docs. For this rather niche use case that seems find and I appreciate you pointing it out!

So in summary, in this case, the `Serializer` in this use case is never actually passed data at runtime, so `self` etc and those capabilities are not invoked.

Happy to close this close this issue as resolved if you are. :)

**Comment by johnthagen** [comment]
> To me it seemed logical to design it like that. 

In hindsight, I do really like this design! It was just I missed that using an _instance_ at all was an option when I read the higher level docs section. That could be on me! 😄

**Comment by johnthagen** [comment]
The last thing to add as I rollback my memory is that I was probably remembering how `drf-spectacular` and `drf-yasg` use type annotations at least a little on things like Serializer Methods, right? So I might have then translated that to thinking perhaps type annotations might work in this instance too.

**Comment by tfranzel** [comment]
> The last thing to add as I rollback my memory

:laughing: 

> drf-spectacular and drf-yasg use type annotations at least a little on things like Serializer Methods, right? 

true... this would have sort of worked as part of a type hint (e.g. `SerializerMethodField`). not sure though given it is a serializer and not a "basic" type but I get your reasoning. we do definitely support a lot more than yasg there.

fyi: we do have `inline_serializer()` helper for shorter one-off serializers.

fyi2: if you JSON is already typed with Pydantic, you may get a schema there and inject it into spectacular as raw schema. Though this is JSON schema and not OpenAPI, and until 3.1 this may not work 100%.
